### PR TITLE
feat: clone official api plugins command

### LINKED ISSUE #68: Create "create-local" command for api that makes local copies of all the plugins
Create "create-local" command for api that makes local copies (clones) of all the plugins and changes them in plugins.json so that you can do development on "stock" plugins using the CLI

### PR DESCRIPTION
**Pull request by tedraykov**
Resolves: #68
 
Issue #68 proposes the name `create-local` for the command to clone all official plugins locally, but in my opinion `clone-api-plugins` is more clear. That name was used in the `reaction-development-platform`.

## Test
- Navigate to an `api` project
- `reaction clone-api-plugins`

Signed-off-by: tedraykov <tedraykov@gmail.com>

### CODE REVIEW

File: commands/clone-api-plugins.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    fs.mkdirSync(apiPluginsDir);
+  }
+}
+
+/**
+ * @summary Checks if plugin is already cloned
+ * @param {String} plugin - key of the plugin
+ * @return {boolean} - true if plugin is already cloned
+ */
+function checkIsPluginAlreadyCloned(plugin) {
```
Comment (brent-hoover) [inline commands/clone-api-plugins.js]: Generally if a function returns a boolean the name should be something like `isAlreadyCloned`

File: utils/apiPluginRepos.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+export default {
```
Comment (brent-hoover) [inline utils/apiPluginRepos.js]: I think this data should come directly from the Reaction project, not hardcoded here

**Review by brent-hoover — CHANGES_REQUESTED** [review]
Looks great. Excited to try it. Just a couple of little points

File: commands/clone-api-plugins.js (hunk 1 above)
Comment (tedraykov) [reply commands/clone-api-plugins.js]: I agree with that. I also will rename the utility function `checkIsProjectOfType`.

File: utils/apiPluginRepos.js (hunk 2 above)
Comment (tedraykov) [reply utils/apiPluginRepos.js]: We can dynamically get the list of plugins from the `plugins.json` indeed. The concern I had is we can't map the plugin name with the repository, although we can generate  the URL in the format:
`https://github.com/reactioncommerce/${plugin-name}.git`

Comment (brent-hoover) [reply utils/apiPluginRepos.js]: Yeah, I think that naming standard should work for all the api-plugins

**Review by vanpho93 — APPROVED** [review]
LGTM really.

**Review by vanpho93 — CHANGES_REQUESTED** [review]
As the issues said: "changes them in plugins.json", I don't think you have done that in the PR since the plugins.json still remains?

**Comment by tedraykov** [comment]
Let's address the linking process in a separate PR. As a developer, I might not want to link all cloned API plugins automatically.

**Comment by brent-hoover** [comment]
I would prefer that we do as the original issue says and make the changes in plugins.json. If we want to add a flag to defeat that default behavior we can

**Review by brent-hoover — APPROVED** [review]

**Pull request merged**